pkg/testing/integration/program.go

```diff
@@ -1011,14 +1015,14 @@ func (pt *programTester) testLifeCycleInitialize(dir string) error {
 	return nil
 }
 
-func (pt *programTester) testLifeCycleDestroy(dir string) error {
+func (pt *ProgramTester) TestLifeCycleDestroy() error {
 	// Destroy and remove the stack.
 	fprintf(pt.opts.Stdout, "Destroying stack\n")
 	destroy := []string{"destroy", "--non-interactive", "--skip-preview"}
 	if pt.opts.GetDebugUpdates() {
 		destroy = append(destroy, "-d")
 	}
-	if err := pt.runPulumiCommand("pulumi-destroy", destroy, dir, false); err != nil {
+	if err := pt.runPulumiCommand("pulumi-destroy", destroy, pt.projdir, false); err != nil {
```

Comment:
I'm a little confused by how this works.  My understanding from reading the code is that `testLifeCycleInitAndDestroy` will still call this method. So immediately after the pulumi up is run. The stack will be destroyed. The `SkipStackRemoval` option prevents `pulumi stack rm` from being run below at line 1034.  

The options don't change between the time you run `pt = integration.ProgramTest()` and  `pt.TestLifeCycleDestroy()`. So the stack still doesn't get removed at the end of the test. 

Either way, this destroy statement is still executing in the middle of the test which I would expect to get rid of the stack exports. and cause problems with the `import` portion of your test.

Reply:
Ah, you're right. I think a better option will be to introduce a new flag for skipping TestLifeCycleDestroy.